rnd/autogpt_builder/src/components/history.ts

```diff
@@ -0,0 +1,44 @@
+// history.ts
+
+type Action = {
```

Comment:
For a bit of type safety and to have our intellisense work a bit more for us we can separate out the type here and use a union type for `Action`.

```suggestion
type ActionType =
  | 'ADD_NODE'
  | 'DELETE_NODE'
  | 'ADD_EDGE'
  | 'DELETE_EDGE'
  | 'UPDATE_NODE'
  | 'MOVE_NODE'
  | 'UPDATE_INPUT'
  | 'UPDATE_NODE_POSITION';

type Action = {
  type: ActionType;
  payload: ActionPayload;
  undo: () => void;
  redo: () => void;
};
```